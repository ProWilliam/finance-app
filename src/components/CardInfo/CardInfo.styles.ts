import { StyleSheet } from 'react-native';
import theme from '../../theme/theme';

const StyleListFinance = StyleSheet.create({

  containerCard : {
    width: `${theme.withScreen.medium}%`,
  },
  titleCard: {
    marginVertical: 40,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',

  },
  iconCard: {
    color: theme.light.tertiary
  },
  bodyCard: {
    padding: 10,

  },
  textTitle: {
    fontSize: theme.size.fontMedium,
    fontWeight: 'bold'
  },
  sectionLogo:{
    marginVertical:30,
  },
  imageCard: {
    width: 200,
    height: 113,
    borderRadius: 4,
    marginLeft: 80
  }
});

export default StyleListFinance;