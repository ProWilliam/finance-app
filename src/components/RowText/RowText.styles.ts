import { StyleSheet } from 'react-native';
import theme from '../../theme/theme';

const StyleRowText = StyleSheet.create({
  rowInfo: {
    marginVertical: 10,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  text: {
    color: theme.light.quintary,
    fontSize:theme.size.fontSmaller
  }
});

export default StyleRowText;