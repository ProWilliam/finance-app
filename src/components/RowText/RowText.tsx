import React from 'react'
import { View, Text } from 'react-native'

// Styles and Types
import styles from './RowText.styles'
import RowTextProps from '../../types/components/RowText.types';


const StyledText: React.FC<RowTextProps> = ({ title, children}) => {
  
  return (
    <View style={styles.rowInfo}>
      <Text testID='title-text' style={styles.text}>{title}</Text>
      <Text testID='content-text' style={styles.text}>{children}</Text>
    </View>
  )
}
export default StyledText;
