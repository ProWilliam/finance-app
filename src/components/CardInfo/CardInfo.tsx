import React, { useState } from 'react';
import { Image ,View, Text, TouchableOpacity } from 'react-native';

// Hook and Components
import useAppNavigation from '../../hooks/useAppNavigation';
import IconClose from '../../assets/icons/IconClose';
import RowText from '../../components/RowText/RowText';

// Style and Types
import styles from './CardInfo.styles';
import CardInfoProps from '../../types/components/CardInfo.types'

const CardInfo: React.FC<CardInfoProps> = ({id, info_text, name, description, logo, date_release, date_revision}) => {

  const [route, setRoute] = useState('');
  const { appNavigation } = useAppNavigation();

  appNavigation(route)
  
  const goBack = () => {
    setRoute('home');
  }

  return(
    <View style={styles.containerCard}>
      <View style={styles.titleCard}>
        <View>
          <Text style={styles.textTitle}>ID:{id}</Text>
          <Text>{info_text}</Text>
        </View>
        <TouchableOpacity
          onPress={goBack}
        >
          <IconClose style={styles.iconCard}/>
        </TouchableOpacity>
      </View>
      <View style={styles.bodyCard}>
        <RowText title='Nombre:'>{name}</RowText>
        <RowText title='Description:'>{description}</RowText>
        <View style={styles.sectionLogo}>
          <Text>Logo:</Text>
          <Image style={styles.imageCard} source={{ uri: logo }} />
        </View>
        <RowText title='Fecha liberación:'>{date_release}</RowText>
        <RowText title='Fecha revisión:'>{date_revision}</RowText>  
      </View>
    </View>
  )
}
export default CardInfo;