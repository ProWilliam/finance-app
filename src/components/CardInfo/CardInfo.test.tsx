import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { Text, TouchableOpacity, Image } from 'react-native';

import CardInfo from './CardInfo'; 
import RowText from '../RowText/RowText';
import styles from './CardInfo.styles';

// Mock de los datos
const mockProps = {
  id: '1', 
  info_text: 'Informacion extra', 
  name: 'Cart black', 
  description: 'Descripcion registrada del producto', 
  logo: 'https://www.visa.com.ec/dam/VCOM/regional/lac/SPA/Default/Pay%20With%20Visa/Tarjetas/visa-signature-400x225.jpg', 
  date_release: '2025-03-02', 
  date_revision: '2026-08-15'
};

// Mock useAppNavigation hook
const mockAppNavigation = jest.fn();

jest.mock('../../hooks/useAppNavigation', () => ({
  __esModule: true,
  default: () => ({
    appNavigation: mockAppNavigation,
  }),
}));

describe('<CardInfo />', () => {

  it('renders correctly with given props', () => {
    const component = renderer.create(<CardInfo {...mockProps} />);
    const tree = component.toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('renders text correctly', () => {
    const component = renderer.create(<CardInfo {...mockProps} />);
    const testInstance = component.root;
    
    // Verify that the ID text, info_text, name, description and dates are rendered.
    const texts =  testInstance.findAllByType(Text);

    // Element ID is Array for this Changed to string.
    const textID = texts.map(text => text.props.children)[0].join('');
    
    expect(textID).toBe('ID:1');
    expect(testInstance.findAllByType(Text)[1].props.children).toBe('Informacion extra');
    expect(testInstance.findAllByType(RowText)[0].props.children).toBe('Cart black');
    expect(testInstance.findAllByType(RowText)[1].props.children).toBe('Descripcion registrada del producto');
    expect(testInstance.findAllByType(RowText)[2].props.children).toBe('2025-03-02');
    expect(testInstance.findAllByType(RowText)[3].props.children).toBe('2026-08-15');
  });

  it('renders the logo image with correct URI', () => {
    const component = renderer.create(<CardInfo {...mockProps} />);
    const testInstance = component.root;
    
    // Verify image URI
    const image = testInstance.findByType(Image);
    expect(image.props.source.uri).toBe(mockProps.logo);
  });

  it('applies correct styles', () => {
    const component = renderer.create(<CardInfo {...mockProps} />);
    const testInstance = component.root;
    
    // Verificar el estilo del contenedor
    const container = testInstance.findByProps({ style: styles.containerCard });
    expect(container).toBeTruthy();
    
    // Verificar el estilo del icono
    const icon = testInstance.findByProps({ style: styles.iconCard });
    expect(icon).toBeTruthy();
  });

  it('calls goBack function on pressing IconClose', () => {
    const component = renderer.create(<CardInfo {...mockProps} />);
  
    act(() => {
      component.root.findByType(TouchableOpacity).props.onPress();
    });

    // Verify that the navigation function was called with "home"
    expect(mockAppNavigation).toHaveBeenCalledWith('home');
  });
});