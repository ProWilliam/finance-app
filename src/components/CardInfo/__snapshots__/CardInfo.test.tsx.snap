// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardInfo /> renders correctly with given props 1`] = `
<View
  style={
    {
      "width": "80%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginVertical": 40,
      }
    }
  >
    <View>
      <Text
        style={
          {
            "fontSize": 34,
            "fontWeight": "bold",
          }
        }
      >
        ID:
        1
      </Text>
      <Text>
        Informacion extra
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        aria-hidden="true"
        bbHeight={50}
        bbWidth={50}
        fill="none"
        focusable={false}
        height={50}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "color": "#eaeaea",
            },
            {
              "flex": 0,
              "height": 50,
              "width": 50,
            },
          ]
        }
        tintColor="#eaeaea"
        vbHeight={24}
        vbWidth={24}
        width={50}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M6 18 17.94 6M18 18 6.06 6"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
  <View
    style={
      {
        "padding": 10,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#6a6a6a",
            "fontSize": 14,
          }
        }
        testID="title-text"
      >
        Nombre:
      </Text>
      <Text
        style={
          {
            "color": "#6a6a6a",
            "fontSize": 14,
          }
        }
        testID="content-text"
      >
        Cart black
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#6a6a6a",
            "fontSize": 14,
          }
        }
        testID="title-text"
      >
        Description:
      </Text>
      <Text
        style={
          {
            "color": "#6a6a6a",
            "fontSize": 14,
          }
        }
        testID="content-text"
      >
        Descripcion registrada del producto
      </Text>
    </View>
    <View
      style={
        {
          "marginVertical": 30,
        }
      }
    >
      <Text>
        Logo:
      </Text>
      <Image
        source={
          {
            "uri": "https://www.visa.com.ec/dam/VCOM/regional/lac/SPA/Default/Pay%20With%20Visa/Tarjetas/visa-signature-400x225.jpg",
          }
        }
        style={
          {
            "borderRadius": 4,
            "height": 113,
            "marginLeft": 80,
            "width": 200,
          }
        }
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#6a6a6a",
            "fontSize": 14,
          }
        }
        testID="title-text"
      >
        Fecha liberación:
      </Text>
      <Text
        style={
          {
            "color": "#6a6a6a",
            "fontSize": 14,
          }
        }
        testID="content-text"
      >
        2025-03-02
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#6a6a6a",
            "fontSize": 14,
          }
        }
        testID="title-text"
      >
        Fecha revisión:
      </Text>
      <Text
        style={
          {
            "color": "#6a6a6a",
            "fontSize": 14,
          }
        }
        testID="content-text"
      >
        2026-08-15
      </Text>
    </View>
  </View>
</View>
`;
